src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
+    /// which contains document metadata as well as the `encrypted_data`, which is the only thing that must be passed
+    /// to [document_decrypt()](trait.DocumentOps.html#tymethod.document_decrypt) in order to decrypt the document.
+    ///
+    /// # Arguments
+    /// - `document_data` - Bytes of the document to encrypt
+    /// - `encrypt_opts` - Document encryption parameters. Default values are provided with `DocumentEncryptOpts::default()`.
     ///
-    /// # Returns
-    /// `DocumentListResult` struct with vec of metadata about each document the user can decrypt.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// use ironoxide::document::DocumentEncryptOpts;
+    /// let data = "secret data".as_bytes();
+    /// let encrypted = sdk.document_encrypt(data, &DocumentEncryptOpts::default()).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_encrypt(
+        &self,
+        document_data: &[u8],
+        encrypt_opts: &DocumentEncryptOpts,
+    ) -> Result<DocumentEncryptResult>;
+
+    /// Decrypts an IronCore encrypted document.
+    ///
+    /// Returns details about the document as well as its decrypted bytes.
+    ///
+    /// # Arguments
+    /// - `encrypted_document` - Bytes of encrypted document. These should be the same bytes returned from
+    /// [document_encrypt()](trait.DocumentOps.html#tymethod.document_encrypt).
+    ///
+    /// # Errors
+    /// Fails if passed malformed data or if the calling user does not have sufficient access to the document.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let encrypted_data: Vec<u8> = vec![];
+    /// let decrypted_document = sdk.document_decrypt(&encrypted_data).await?;
+    /// let decrypted_data = decrypted_document.decrypted_data();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
+
+    /// Lists all of the IronCore encrypted documents that the calling user has been
+    /// granted access to.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # use ironoxide::document::DocumentListMeta;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// let document_data = sdk.document_list().await?;
+    /// let documents: Vec<DocumentListMeta> = document_data.result().to_vec();
+    /// # Ok(())
+    /// # }
     async fn document_list(&self) -> Result<DocumentListResult>;
 
-    /// Get the metadata for a specific document given its ID.
+    /// Returns the metadata for an IronCore encrypted document.
```

Comment:
```suggestion
    /// Returns the metadata for an encrypted document. Does not return the encrypted document bytes.
```

Reply:
I think I'll go with
```
/// Returns the metadata for an encrypted document.
///
/// This will not return the encrypted document bytes, as they are not stored by IronCore.
```